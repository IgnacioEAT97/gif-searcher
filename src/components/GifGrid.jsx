import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';
export const GifGrid = ({category}) => {

  const {images,isLoading} = useFetchGifs(category);


//Abajo implementaremos un custoum hook para aplicar DRY(dont reapeat yourself)
  /* const [images, setImages] = useState([]);
  const getImages = async () => {
    const newImages = await getGifs(category);
    setImages(newImages);
    //console.log(images);
  }
  useEffect(() => {
    getImages();
  },[])
   */
  return (
    <>
    <h3>{category}</h3>
    {
      isLoading && (<h1> Cargando </h1>)
    }
    <div className="card-grid">
      {images.map((image) => 
        /* (<li key={id}> {title}</li>) */
        <GifItem
         key={image.id}
         {...image} // Esparce todas las props del padre al hijo
         /> 
      )}
    </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
}