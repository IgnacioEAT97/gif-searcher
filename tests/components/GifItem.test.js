import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Probando GifItem', () => { 
    const title = 'Kakaroto'
    const url = 'https://kakaroto/goku.jpg'
    test('debe generar snapshot', () => {

        const {container} = render(<GifItem title={title} url={url}></GifItem>)
        expect(container).toMatchSnapshot();
     })

     test('debe de mostrar la imagen con el URL y el ALT indicado', () => { 
        render(<GifItem title={title} url={url}></GifItem>)
        /* expect(screen.getByRole('img').src).toBe(url)
        expect(screen.getByRole('img').alt).toBe(title) */
        const {src,alt} = screen.getByRole('img'); // Es mejor desestructurarlos y mas ordenado
        expect(src).toBe(url)
        expect(alt).toBe(title);
        
      })

      test('debe de mostrar el titulo en el compoentne', () => { 
        render(<GifItem title={title} url={url}></GifItem>)
        expect(screen.getByText(title)).toBeTruthy() // Busca el texto y NO EL ATRIBUTO
       })
 });