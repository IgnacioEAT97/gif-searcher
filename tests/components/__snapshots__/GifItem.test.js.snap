// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando GifItem debe generar snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Kakaroto"
      src="https://kakaroto/goku.jpg"
    />
    <p>
      Kakaroto
    </p>
  </div>
</div>
`;
